augur/curate/format_dates.py

```diff
@@ -222,27 +280,128 @@ def run(args, records):
                 if failure_reporting is DataErrorMethod.ERROR_FIRST:
                     raise AugurError(dedent(f"""\
                         {failure_message}
-                        {failure_suggestion}"""))
+                        {format_failure_suggestion}"""))
 
                 if failure_reporting is DataErrorMethod.WARN:
                     print_err(f"WARNING: {failure_message}")
 
                 # Keep track of failures for final summary
-                failures.append((record_id, field, date_string))
+                format_failures.append((record_id, field, date_string))
             else:
                 record[field] = formatted_date_string
 
-        yield record
+        return record
+
+    def apply_bounds(record, record_id):
+        record = record.copy()
 
-    if failure_reporting is not DataErrorMethod.SILENT and failures:
-        if failure_reporting is DataErrorMethod.ERROR_ALL:
-            raise AugurError(dedent(f"""\
-                Unable to format dates for the following (record, field, date string):
-                {indented_list(map(repr, failures), "                ")}
-                {failure_suggestion}"""))
+        # Only modify empty dates if both bounds are set
+        if record[args.target_date_field] == "XXXX-XX-XX" and not (args.target_date_field_min and args.target_date_field_max):
+            return record
+
+        original_date = get_numerical_date_from_value(record[args.target_date_field], fmt="%Y-%m-%d")
+
+        # Keep exact dates as-is
+        # Maybe worth adding a warning if the date is out of bounds?
```

Comment:
Given the errors in the cases where an incomplete date is out-of-bounds, I would be surprised that an out-of-bounds complete date didn't provoke similar behavior. I think a warning is a minimum, and would ask, why not an error?

Reply:
+1 for error/warning if out of bounds even if the date is an exact date. 

(Made me think that it would be useful to have a option such as `--target-max-date today` to warn about future dates in the metadata.)

Reply:
Good point. I'll update to error in the new PR.